### PR TITLE
Script to classify charts

### PR DESCRIPTION
**Pull request by bkamapantula**

### CODE REVIEW

File: classify.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+from skimage.color import rgba2rgb
+from skimage.transform import resize
+from skimage.io import imread
+from sklearn.preprocessing import LabelEncoder
+from tensorflow import keras
+from gramex.data import filter as gfilter
+
+
+OP = os.path
+FOLDER = OP.abspath(OP.dirname(__file__))
```
Comment (jaidevd) [inline classify.py]: I think `abspath` is redundant here. `dirname` already gives the absolute path.

File: classify.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    # load model and predict
+    model_path = OP.join(FOLDER, 'assets', 'models', model_name)
+    model = keras.models.load_model(model_path)
+    predictions = model.predict(image)
+
+    # fetch the label
+    charts = gfilter(chart_db, table='charts')
+    to_keep = charts['validated_label'].value_counts()[
+        charts['validated_label'].value_counts() > 21].index
+    charts = charts[charts['validated_label'].isin(to_keep)]
```
Comment (jaidevd) [inline classify.py]: The above four lines can be pre-computed. We may not want this filtering to happen on every call of this function.

Comment (bkamapantula) [reply classify.py]: fixed via 727c457

File: classify.py (hunk 1 above)
Comment (bkamapantula) [reply classify.py]: fixed via 727c457

File: classify.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    image = decodebytes(image.encode())
+    image = imread(BytesIO(image))
+
+    image = rgba2rgb(image)
+    image = resize(image, (224, 224), preserve_range=True)
+    image = image.reshape((1, ) + image.shape)
+
+    # load model and predict
+    model_path = OP.join(FOLDER, 'assets', 'models', model_name)
+    model = keras.models.load_model(model_path)
```
Comment (jaidevd) [inline classify.py]: Please replace this with `gramex.cache.open(model_path, keras.models.load_model)`

File: classify.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    predictions = model.predict(image)
+
+    lenc = LabelEncoder()
+    lenc.fit_transform(charts['validated_label'])
```
Comment (jaidevd) [inline classify.py]: Let's move this out too.

**Review by jaidevd — CHANGES_REQUESTED** [review]

**Pull request merged**